Connectors/Rhino/Speckle.Connectors.GrasshopperShared/Parameters/Wrappers/SpeckleDataObjectWrapper.cs

```diff
@@ -126,8 +128,76 @@ public void DrawPreviewRaw(DisplayPipeline display, DisplayMaterial material)
     }
   }
 
-  // TODO: CNX-2095
-  // public virtual void Bake()
+  /// <summary>
+  /// Bakes the DataObject as a Rhino group containing all the display geometries.
+  /// </summary>
+  /// <param name="doc">Rhino doc to bake into</param>
+  /// <param name="objIds">Collection to store created objects GUIDs</param>
+  /// <param name="bakeLayerIndex">Layer index to bake geometries to (-1 for automatic)</param>
+  /// <param name="layersAlreadyCreated">Indicates whether layers have already been created or not</param>
+  /// <param name="baseLayerName">Base layer name for group naming</param>
+  public virtual void Bake(
+    RhinoDoc doc,
+    List<Guid> objIds,
+    int bakeLayerIndex = -1,
+    bool layersAlreadyCreated = false,
+    string? baseLayerName = null
+  )
+  {
+    // handles layer creation (if needed)
+    if (!layersAlreadyCreated && bakeLayerIndex < 0 && Path.Count > 0 && Parent != null)
+    {
+      bakeLayerIndex = Parent.Bake(doc, objIds, false);
+      if (bakeLayerIndex < 0)
+      {
+        return; // failed to create layers
+      }
+    }
+
+    if (Geometries.Count == 0)
+    {
+      return; // nothing to bake
+    }
+
+    // bake all display geometries as individual objects
+    List<Guid> geometryIds = [];
+
+    foreach (SpeckleGeometryWrapper geometryWrapper in Geometries)
+    {
+      if (geometryWrapper.GeometryBase != null)
+      {
+        // geometry wrappers should already be synced via prop setters, assume we're in a consistent state
+        List<Guid> currentGeometryIds = [];
+        geometryWrapper.Bake(doc, currentGeometryIds, bakeLayerIndex, true);
+        geometryIds.AddRange(currentGeometryIds);
+      }
+    }
+
+    // create a group for all geometries
+    if (geometryIds.Count > 0)
+    {
```

Comment:
hmm I don't think we'd want to create a group of 1 object

Reply:
[fix: pr comment](https://github.com/specklesystems/speckle-sharp-connectors/pull/974/commits/831ac5c5215ff50a03f469e928242631c347d7c6)